sdk/build.gradle

```diff
@@ -166,60 +165,65 @@ subprojects {
-                        url = ext.siteUrl
-
-                        // Set identifiers of assemble.
-                        groupId = ext.groupId
-                        artifactId = project.name
-
-                        // Set license information.
-                        licenses {
-                            license {
-                                name = ext.licenseName
-                                url = ext.licenseSite
-                            }
+    publishing {
+        repositories {
+            maven {
+                url = ext.mavenRepoUrl
+                credentials {
+                    username = ext.mavenUser
+                    password = ext.mavenKey
+                }
+            }
+        }
+        publications {
+            release(MavenPublication) {
+                pom {
+
+                    // Set base information about assemble.
+                    packaging = 'aar'
+                    name = project.name
+                    description = project.description
+                    url = ext.siteUrl
+
+                    // Set identifiers of assemble.
+                    groupId = ext.groupId
+                    artifactId = project.name
+
+                    // Set license information.
+                    licenses {
+                        license {
+                            name = ext.licenseName
+                            url = ext.licenseSite
                         }
+                    }
 
-                        // Set information about developers.
-                        developers {
-                            developer {
-                                id = ext.developerId
-                                name = ext.developerName
-                                email = ext.developerEmail
-                            }
+                    // Set information about developers.
+                    developers {
+                        developer {
+                            id = ext.developerId
+                            name = ext.developerName
+                            email = ext.developerEmail
                         }
+                    }
 
-                        // Set information about connection with developers.
-                        scm {
-                            connection = ext.gitUrl
-                            developerConnection = ext.gitUrl
-                            url = ext.siteUrl
-                        }
+                    // Set information about connection with developers.
+                    scm {
+                        connection = ext.gitUrl
+                        developerConnection = ext.gitUrl
+                        url = ext.siteUrl
                     }
                 }
-            }
 
-            signing {
-                required { gradle.taskGraph.hasTask("uploadArchives") }
-                sign configurations.archives
+                afterEvaluate {
```

Comment:
Do we really need to execute "`from components.release`" in afterEvaluate callback? Couldn't we do it inside "`publications`"?

Reply:
It is possible, but it might not be the best practice. afterEvaluate ensures that all dependencies have been built and evaluated, thus calling components.release inside afterEvaluate helps us ensure that all the configurations and dependencies have been set before the release starts. 'publications' is merely used to define the libraries/artefacts needed for the release.